spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AbstractActuatorTemplate.java

```diff
@@ -0,0 +1,104 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.app;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public abstract class AbstractActuatorTemplate implements ActuatorOperations {
+
+	protected final Log logger = LogFactory.getLog(getClass().getName());
+
+	protected final RestTemplate restTemplate;
+	protected final AppDeployer appDeployer;
+
+	protected AbstractActuatorTemplate(RestTemplate restTemplate, AppDeployer appDeployer) {
+		this.restTemplate = restTemplate;
+		this.appDeployer = appDeployer;
+	}
+
+	@Override
+	public <T> T getFromActuator(String deploymentId, String guid, String endpoint, Class<T> responseType) {
+		String actuatorUrl = getActuatorUrl(deploymentId, guid);
+
+		ResponseEntity<T> responseEntity = get(UriComponentsBuilder
+				.fromHttpUrl(actuatorUrl).path(normalizePath(endpoint)).toUriString(), responseType);
+		if (responseEntity.getStatusCode().isError()) {
+			logger.error(responseEntity.getStatusCode().toString());
+		}
+		return responseEntity.getBody();
+	}
+
+	@Override
+	public <T, R> R postToActuator(String deploymentId, String guid, String endpoint, T body,
+			Class<R> responseType) {
+		String actuatorUrl = getActuatorUrl(deploymentId, guid);
+		ResponseEntity<R> responseEntity = post(UriComponentsBuilder
+				.fromHttpUrl(actuatorUrl).path(normalizePath(endpoint)).toUriString(), body, responseType);
+		if (responseEntity.getStatusCode().isError()) {
+			logger.error(responseEntity.getStatusCode().toString());
+		}
+		return responseEntity.getBody();
+	}
+
+	protected HttpHeaders requestHeaders() {
+		HttpHeaders headers = new HttpHeaders();
+		headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+		headers.setContentType(MediaType.APPLICATION_JSON);
+		return headers;
+	}
+
+	private String normalizePath(String path) {
+		return path.startsWith("/") ? path : "/" + path;
+	}
+
+	private <T> ResponseEntity<T> get(String url, Class<T> responseType) {
+		return restTemplate.exchange(url, HttpMethod.GET,
+				new HttpEntity(requestHeaders()), responseType);
+	}
+
+	private <T, R> ResponseEntity<R> post(String url, T requestBody, Class<R> responseType) {
+		HttpEntity<T> requestEntity = new HttpEntity(requestBody, requestHeaders());
```

Comment:
**[TinyNit]** Inconsistent w/ its `get` counterpart in that one inlines the `HttpEntity`.